drivers/regulator/regulator_max20335.c

```diff
@@ -49,11 +52,15 @@ struct regulator_max20335_desc {
 	const struct linear_range *ua_range;
 };
 
+struct regulator_max20335_common_config {
+	struct i2c_dt_spec bus;
+};
+
 struct regulator_max20335_config {
 	struct regulator_common_config common;
-	struct i2c_dt_spec bus;
 	const struct regulator_max20335_desc *desc;
 	uint8_t source;
+	const struct device *parent;
```

Comment:
note: not sure you save much by doing this (I removed it from some code a while ago), as it complicates things for just an address byte, depending on alignment you may actually save nothing.

Reply:
I followed the same pattern as in `pca9420`. Not sure what savings you have in mind?

Reply:
Both the common and the individual regulator configs can have their own copy of `struct i2c_dt_spec bus`.
Then you can access `->bus` directly instead of having to go via the parent.

Reply:
As you can see, `struct i2c_dt_spec` has been removed from the individual regulator configs. Am I still wrong, am I still blind?

Reply:
I believe Gerard is suggesting that you keep `struct i2c_dt_spec bus` in both the common and individual configs.
Then you don't need to get the parent config pointer, you can keep the original code.

Reply:
Thanks, I've fixed that.